huaweicloud/services/nat/resource_huaweicloud_nat_private_gateway.go

```diff
@@ -0,0 +1,229 @@
+package nat
+
+import (
+	"context"
+	"log"
+	"regexp"
+
+	"github.com/hashicorp/go-multierror"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+
+	"github.com/chnsz/golangsdk/openstack/nat/v3/gateways"
+
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/common"
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/config"
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/utils"
+)
+
+const (
+	PrivateSpecTypeSmall      string = "Small"
+	PrivateSpecTypeMedium     string = "Medium"
+	PrivateSpecTypeLarge      string = "Large"
+	PrivateSpecTypeExtraLarge string = "Extra-Large"
+)
+
+func ResourcePrivateGateway() *schema.Resource {
+	return &schema.Resource{
+		CreateContext: resourcePrivateGatewayCreate,
+		ReadContext:   resourcePrivateGatewayRead,
+		UpdateContext: resourcePrivateGatewayUpdate,
+		DeleteContext: resourcePrivateGatewayDelete,
+
+		Importer: &schema.ResourceImporter{
+			StateContext: schema.ImportStatePassthroughContext,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"region": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Computed:    true,
+				ForceNew:    true,
+				Description: "The region where the private NAT gateway is located.",
+			},
+			"subnet_id": {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "The network ID of the subnet to which the private NAT gateway belongs.",
+			},
+			"name": {
+				Type:     schema.TypeString,
+				Required: true,
+				ValidateFunc: validation.All(
+					validation.StringMatch(regexp.MustCompile("^[\u4e00-\u9fa5\\w-]*$"),
```

Comment:
```suggestion
					validation.StringMatch(regexp.MustCompile(`^[\u4e00-\u9fa5\w-]*$`),
```

Reply:
panic: regexp: Compile(`^[\u4e00-\u9fa5\w-]*$`): error parsing regexp: invalid escape sequence: `\u`

Reply:
Okey, `\x{4E00}-\x{9FFC}` is fine.